enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -45,6 +45,35 @@ extern "C" {
 
 pub enum Context {}
 
+// Helper for allocation of untrusted memory in a Block.
+pub struct Cursor<'a>(Option<&'a mut [u8]>);
+
+impl<'a> Cursor<'a> {
+    pub fn new(buffer: &'a mut [u8]) -> Self {
+        Self(Some(buffer))
+    }
+
+    pub fn alloc<T>(&mut self, count: usize) -> Result<&'a mut [T], ()> {
+        let buf = self.0.take().unwrap();
+
+        let mid = {
+            let (padding, data, _) = unsafe { buf.align_to_mut::<T>() };
+
+            if data.len() < count {
+                return Err(());
+            }
+
+            padding.len() + size_of::<T>() * count
+        };
+
+        let (data, next) = buf.split_at_mut(mid);
+
+        self.0.replace(next);
+
+        Ok(unsafe { data.align_to_mut::<T>().1 })
+    }
+}
```

Comment:
See my comments about the cursor in the other PR. Note that cursor should probably be its own PR (and should probably have tests).